portal/src/components/roles-and-groups/GroupsEmptyView.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { FormattedMessage } from "@oursky/react-messageformat";
+import { useParams } from "react-router-dom";
+
+import { RolesAndGroupsEmptyView } from "./RolesAndGroupsEmptyView";
+
+import iconGroups from "../../images/material-groups.svg";
+import ReactRouterLink from "../../ReactRouterLink";
+
+export const GroupsEmptyView: React.VFC<{ className?: string }> =
+  function GroupsEmptyView({ className }) {
+    const { appID } = useParams<{ appID: string }>();
+
+    return (
+      <RolesAndGroupsEmptyView
+        className={className}
+        icon={<img src={iconGroups} />}
+        title={<FormattedMessage id="GroupsEmptyView.title" />}
+        description={<FormattedMessage id="GroupsEmptyView.description" />}
+        button={
+          <ReactRouterLink.CustomComponent
+            component={RolesAndGroupsEmptyView.CreateButton}
+            to={`/project/${appID}/user-management/groups/add-group`}
+            text={<FormattedMessage id="GroupsEmptyView.button.text" />}
+          />
```

Comment:
Not sure if we want to do it this way, seems introducing a redundant layer to me. 
At least I think a common usage would be wrapping `ReactRouterLink.CustomComponent`  inside `ReactRouterLink`?

Reply:
> At least I think a common usage would be wrapping ReactRouterLink.CustomComponent inside ReactRouterLink?

Why we will use it in this way? When using `ReactRouterLink.CustomComponent`, we don't need `ReactRouterLink`.

In fact, I introduced `ReactRouterLink.CustomComponent` because the `component` prop of `ReactRouterLink` doesn't have a correct typing. It assumed you must pass in a component which have same props of a `<a>`, which make it difficult to use.

So I separated `ReactRouterLink.CustomComponent` from `ReactRouterLink`. `ReactRouterLink` now only renders to `<a>`. And `ReactRouterLink.CustomComponent` only renders to your provided `component`, and it accepts all props your component accepts.

Reply:
If that so, sounds like `ReactRouterLink` is not needed anymore as well.

> When using ReactRouterLink.CustomComponent, we don't need ReactRouterLink

What I mean is I think there is a common convention that we wrap component `X.Y` inside `X`? 
For example
```
<Button>
  <Button.Label>
  </Button.Label>
</Button>
```

Reply:
The problem is it will be difficult to make a type safe interface for `as="a"` or `as="button"`. So I didn't support it.